src/plugins/template/src/infer_request.cpp

```diff
@@ -0,0 +1,253 @@
+    for (size_t i = 0; i < m_input_tensors.size(); i++) {
+        // No ROI extraction is needed
+        m_plugin_input_tensors[i] = get_template_model()->get_template_plugin()->_backend->create_tensor(
+            m_input_tensors.at(i).get_element_type(),
+            m_input_tensors.at(i).get_shape(),
+            m_input_tensors.at(i).data());
+    }
+    OPENVINO_ASSERT(m_output_tensors.size() == m_plugin_output_tensors.size());
+    for (size_t i = 0; i < m_output_tensors.size(); i++) {
+        m_plugin_output_tensors[i] = get_template_model()->get_template_plugin()->_backend->create_tensor(
+            m_output_tensors.at(i).get_element_type(),
+            m_output_tensors.at(i).get_shape(),
+            m_output_tensors.at(i).data());
+    }
+    m_durations[Preprocess] = Time::now() - start;
+}
+// ! [infer_request:infer_preprocess]
+
+// ! [infer_request:start_pipeline]
+void TemplatePlugin::InferRequest::start_pipeline() {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, m_profiling_task[StartPipeline])
+    auto start = Time::now();
+    m_executable->call(m_plugin_output_tensors, m_plugin_input_tensors);
+    m_durations[StartPipeline] = Time::now() - start;
+}
+// ! [infer_request:start_pipeline]
+
+void TemplatePlugin::InferRequest::wait_pipeline() {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, m_profiling_task[WaitPipeline])
+    auto start = Time::now();
+    // TODO: Wait pipeline using driver API or other synchronizations methods
+    // NOTE: not used in current implementation since `startPipeline` executes pipiline synchronously
+    m_durations[WaitPipeline] = Time::now() - start;
+}
+
+// ! [infer_request:infer_postprocess]
+void TemplatePlugin::InferRequest::infer_postprocess() {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, m_profiling_task[Postprocess]);
+    auto start = Time::now();
+    // We don't need post-processing in template plugin
+    m_durations[Postprocess] = Time::now() - start;
+}
+// ! [infer_request:infer_postprocess]
+
+// ! [infer_request:get_blob]
+ov::Tensor TemplatePlugin::InferRequest::get_tensor(const ov::Output<const ov::Node>& port) const {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, "get_tensor");
+    auto found_port = find_port(port);
+    OPENVINO_ASSERT(found_port.found(), "Failed to get tensor, port ", port, " cannot be found.");
+
+    ov::Tensor data;
+    if (found_port.is_input()) {
+        data = m_input_tensors[found_port.idx];
+        ov::Shape shape;
+        if (!data.get_size()) {
+            auto&& parameters = get_template_model()->m_model->get_parameters();
+            const auto& pshape = parameters.at(found_port.idx)->get_partial_shape();
+            shape = pshape.is_dynamic() ? ov::Shape{0} : pshape.get_shape();
+            auto this_non_const = const_cast<InferRequest*>(this);
+            AllocateImplSingle(this_non_const->m_input_tensors, found_port.idx, port.get_element_type(), shape);
```

Comment:
maybe we don't need `const` for `get_tensor` then?
Looking at this code, I suppose it's OK that plugin perform some non-const operations in this method

Reply:
I removed this code from get_tensor :)